templates/zerver/api/changelog.md

```diff
@@ -0,0 +1,78 @@
+below features are supported.
+
+## Changes in Zulip 2.2
+
+**Not yet stabilized**:
+
+* [`POST /register`](/api/register-queue): Added `slim_presence`
+  parameter.  Still under development.
+
+**Feature level 2**:
+
+* [`POST /messages/{message_id}/reactions`](/api/add-emoji-reaction):
+  The `emoji_type` parameter is optional; the server will guess the
+  emoji_type if it is not specified (checking custom emoji, then
+  unicode emoji for any with the provided name).
+* `reactions` objects returned by the API (both in `GET /messages` and
+  in `GET /events`) now include the user who reacted in a top-level
+  `user_id` field.  The legacy `user` dictionary (which had
+  inconsistent format between those two endpoints) is deprecated.
+
+**Feature level 1**:
+
+* [`GET /server_settings`](/api/server-settings): Added
+  `zulip_feature_level`, which can be used by clients to detect which
+  of the features described in this changelog are supported.
+* [`POST /register`](/api/register-queue): Added `zulip_feature_level`
+  to the response if `zulip_version` is among the requested
+  `event_types`.
+* [`GET /users`](/api/get-all-users): User objects for bots now
+  contain a `bot_owner_id`, replacing the previous `bot_owner` field
+  (which had the email address of the bot owner).
+* [`GET /users/{user_id}`](/api/get-user): Endpoint added.
+* [`GET /messages`](/api/get-messages): Add support for string-format
+  values for the `anchor` parameter, deprecating and replacing the
+  `use_first_unread_anchor` parameter.
+* [`POST /users/me/subscriptions`](/api/add-subscriptions): Replaced
+  `is_announcement_only` boolean with `stream_post_policy` enum for
+  specifying who can post to a stream.
+* [`PATCH /streams/{stream_id}`](/api/update-stream): Replaced
+  `is_announcement_only` boolean with `stream_post_policy` enum for
+  specifying who can post to a stream.
+* [`GET /streams/{stream_id}`](/api/update-stream): Replaced
+  `is_announcement_only` boolean with `stream_post_policy` enum for
+  specifying who can post to a stream.
+* `GET /api/v1/user_uploads`: Added new endpoint for requesting a
+  temporary URL for an uploaded file that does not require
+  authentication to access (e.g. for passing from a Zulip desktop,
+  mobile, or terminal app to the user's default browser).
+
+## Changes in Zulip 2.1
+
+* [`GET /users`](/api/get-all-users): Added `include_custom_profile_fields`
+  to request custom profile field data.
+* [`GET /users/me`](/api/get-profile): Added `avatar_url` field,
+  containing the user's avatar URL, to the response.
+* [`GET /users/me/subscriptions`](/api/get-subscribed-streams): Added
+  `include_subscribers` parameter controlling whether data on the
+  other subscribers is included.  Previous behavior was to always send
+  subscriber data.
+* [`GET /server_settings`](/api/server-settings): Added
+  `external_authentication_methods` structure, used to display login
+  buttons nicely in the mobile apps.
```

Comment:
The `/users/me` and `/server_settings` changes aren't mentioned in those docs pages -- it'd be good to add them.

(Hmm, also most of `/users/me` isn't really documented! Most of the properties could be covered well by a reference to get-user or get-all-users. But I'm not acutally sure what to make of `client_id`, and I haven't gone through the whole list to find what else there is.)

Reply:
Agreed.

We currently don't have a standard system for responses; so part of the reason I'm adding mentions of things like that here is as reminders to document them when we do add the responses data.

Part of our master OpenAPI docs plan is to make the "responses" section for API endpoints work like the "parameters" section, generated from the OpenAPI data and validated by our test suite.